test-framework/src/test/java/org/apache/brooklyn/test/framework/yaml/TestCaseYamlTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.test.framework.yaml;
+
+import org.apache.brooklyn.camp.brooklyn.AbstractYamlRebindTest;
+import org.apache.brooklyn.core.entity.BrooklynConfigKeys;
+import org.apache.brooklyn.entity.machine.MachineEntity;
+import org.apache.brooklyn.entity.stock.BasicApplication;
+import org.apache.brooklyn.test.framework.TestCase;
+import org.apache.brooklyn.test.framework.TestEffector;
+import org.apache.brooklyn.test.framework.TestSensor;
+import org.apache.brooklyn.test.framework.TestSshCommand;
+import org.apache.brooklyn.util.core.internal.ssh.RecordingSshTool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+//Checks that the test cases work in YAML
+@Test
+public class TestCaseYamlTest extends AbstractYamlRebindTest {
+    
+    // TODO Would like this test to be in brooklyn-camp, but that would create a circular dependency.
+    // brooklyn-test-framework depends on brooklyn-camp.
```

Comment:
I assume it depends on `brooklyn-camp` just for the tests? That's the problem with other modules as well - we want the functionality tested in YAML, but the code has nothing to do with CAMP otherwise. That's the reason why `brooklyn-camp`'s tests are huge - they are testing other's modules code through YAML.

I think the way to go is to create a camp-itest module just for tests like this.

On the other hand the YAML is just an alternative spec description so could we just use in-java specs to achieve the same goal?